VideoWeb/VideoWeb/ClientApp/src/app/shared/toast/room-closing/room-closing-toast.component.ts

```diff
@@ -0,0 +1,66 @@
+import * as moment from 'moment';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ToastrService, ToastPackage } from 'ngx-toastr';
+import { Hearing } from '../../models/hearing';
+import { Subscription } from 'rxjs';
+import { VhToastComponent } from '../vh-toast.component';
+import { ClockService } from 'src/app/services/clock.service';
+
+@Component({
+    templateUrl: './room-closing-toast.component.html',
+    styleUrls: ['./room-closing-toast.component.scss']
+})
+export class RoomClosingToastComponent extends VhToastComponent implements OnInit, OnDestroy {
+    durationStr: string;
+    timeLeft$: Subscription;
+    private hearing: Hearing;
+    private expiryTime: Date;
+
+    constructor(protected toastrService: ToastrService, public toastPackage: ToastPackage, private clockService: ClockService) {
+        super(toastrService, toastPackage);
+    }
+
+    ngOnInit(): void {
+        this.timeLeft$ = this.clockService.getClock().subscribe(time => this.calcTimeLeft(time));
+    }
+
+    ngOnDestroy(): void {
+        this.timeLeft$?.unsubscribe();
+    }
+
+    setHearing(hearing: Hearing) {
```

Comment:
this can probably be ignored? 
if set in options similar to vhtoast options it would remove the race conditions in calcTimeLeft

Reply:
Now using "expiry date" instead of "hearing" and passing that through a property setter a-la "vhToastOptions"